### PR TITLE
make pretty-printer more usable

### PR DESCRIPTION
**Pull request by vmchale**

### CODE REVIEW

File: language-plutus-core/src/Data/Text/Prettyprint/Doc/Custom.hs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                                        , (</>)
+                                        , (<//>)
+                                        ) where
+
+import           Data.Text.Prettyprint.Doc
+
+infixr 5 </>
+infixr 5 <//>
+
+(<//>) :: Doc a -> Doc a -> Doc a
```
Comment (michaelpj) [inline language-plutus-core/src/Data/Text/Prettyprint/Doc/Custom.hs]: Haddock all these please.

File: language-plutus-core/.hlint.yaml [hunk 2]
```diff
@@ -1,27 +0,0 @@
----
```
Comment (michaelpj) [inline language-plutus-core/.hlint.yaml]: irrelevant deletions?

Comment (vmchale) [reply language-plutus-core/.hlint.yaml]: Yes, I've fixed this with the new commits.

File: language-plutus-core/src/Language/PlutusCore/TH.hs [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 module Language.PlutusCore.TH (plcTerm, plcType, plcProgram) where
 
-import           Language.Haskell.TH        hiding (Name, Type)
+import           Language.Haskell.TH           hiding (Name, Type)
```
Comment (michaelpj) [inline language-plutus-core/src/Language/PlutusCore/TH.hs]: I thought we agreed not to make random formatting changes to unrelated files?

File: language-plutus-core/src/Language/PlutusCore/Error.hs [hunk 4]
```diff
@@ -13,13 +13,10 @@ import           Language.PlutusCore.Renamer
 import           Language.PlutusCore.TypeSynthesis
 import           PlutusPrelude
 
-import           Data.Text as T
```
Comment (michaelpj) [inline language-plutus-core/src/Language/PlutusCore/Error.hs]: I put this there so I could use it in another PR, please don't delete it again!

File: language-plutus-core/src/Language/PlutusCore/TH.hs (hunk 3 above)
Comment (vmchale) [reply language-plutus-core/src/Language/PlutusCore/TH.hs]: This is `stylish-haskell`.

File: language-plutus-core/src/Language/PlutusCore/Error.hs (hunk 4 above)
Comment (vmchale) [reply language-plutus-core/src/Language/PlutusCore/Error.hs]: This is incorrect and should not have been merged without review. Currently the printer for it prints with `"` on both sides.

File: language-plutus-core/src/Language/PlutusCore/TH.hs (hunk 3 above)
Comment (michaelpj) [reply language-plutus-core/src/Language/PlutusCore/TH.hs]: Yes, but we haven't agreed to use that everywhere yet, and changes like that should happen all at once. Doing it this way just maximizes the amount of conflicts.

File: language-plutus-core/src/Language/PlutusCore/Error.hs (hunk 4 above)
Comment (michaelpj) [reply language-plutus-core/src/Language/PlutusCore/Error.hs]: If you'd like to take it out, then do that *in another PR* and we can talk about it there.

Comment (vmchale) [reply language-plutus-core/src/Language/PlutusCore/Error.hs]: Sounds good, will do.

**Comment by michaelpj** [comment]
I'm going to merge despite the conflicty stuff in `TH.hs`, since the only person that's going to conflict with is me and I'll eat it.
